mitiq/calibration/calibrator.py

```diff
@@ -92,208 +94,118 @@ def get_cost(self) -> Dict[str, int]:
-            ideal_value = distribution[bitstring_to_measure]
-            noisy_error = ideal_value - noisy_value
-
-            mitigated_values: Dict[str, Dict[str, Any]] = {
-                technique.name: {"results": [], "improvement_factor": None}
-                for technique in self.settings.techniques
-            }
-            for strategy in self.settings.make_strategies():
+
+            for strategy in strategies:
                 mitigated_value = strategy.mitigation_function(
                     circuit, expval_executor
                 )
-                improvement_factor = abs(
-                    noisy_error / (ideal_value - mitigated_value)
-                )
 
                 result = {
-                    "mitigated_value": mitigated_value,
-                    "improvement_factor": improvement_factor,
-                    "strategy": strategy,
-                    **strategy.as_dict(),
-                }
-                technique = strategy.technique.name
-                mitigated_values[technique]["results"].append(result)
-
-            circuit_info = problem.problem_summary_dict()
-            expvals.append(
-                {
-                    "circuit_info": circuit_info,
-                    "noisy_value": noisy_value,
-                    "ideal_value": distribution[bitstring_to_measure],
-                    "mitigated_values": mitigated_values,
+                    "mitigated": mitigated_value,
+                    "noisy": noisy_value,
+                    "ideal": problem.largest_probability(),
                 }
-            )
-        return expvals
+                summary = Result(
+                    strategy=strategy, problem=problem, results=result
+                )
+                results.append(summary)
 
-    def compute_improvements(
-        self, experiment_results: List[Dict[str, Any]]
-    ) -> None:
-        """Computes the improvement factors for each calibration circuit that
-        was run. Saves the improvement factors in the input dictionary.
+        return results
 
-        Args:
-            experiment_results: Results obtained from :func:`run_circuits`.
-        """
-        regularizing_epsilon = 1e-30
-        for result in experiment_results:
-            ideal_value = result["ideal_value"]
-            noisy_value = result["noisy_value"]
-            for di in result["mitigated_values"].values():
-                results = di["results"]
-                mitigated_values = [di["mitigated_value"] for di in results]
-                improvement_factor = abs(noisy_value - ideal_value) / sqrt(
-                    regularizing_epsilon
-                    + len(mitigated_values)
-                    * sum(
-                        (mitigated_value - ideal_value) ** 2
-                        for mitigated_value in mitigated_values
-                    )
-                )
-                di["improvement_factor"] = improvement_factor
+    def filter_problems(
```

Comment:
this and other methods right below don't have a docstring

Reply:
Further restructuring resulted in these getting removed.